src/futhark.nim

```diff
@@ -516,6 +516,10 @@ proc createConst(origName: string, node: JsonNode, state: var State, comment: st
           let intNode = newNimNode(nnkIntLit)
           intNode.intVal = node["value"].num
           intNode
+        elif node["value"].kind == JFloat:
+          let floatNode = newNimNode(nnkFloatLit)
+          floatNode.floatVal = node["value"].fnum
+          floatNode
```

Comment:
Hmm, this might be problematic. If the initial float is `0.0f`, then Nim reads it in as a zero float, but the JSON serializer will just output `0` IIRC. So when it is read back the data type is `JInt` and not `JFloat`. But I guess we reinterpret these as their proper C value anyways..

Reply:
It's stored as `0.0` in the opir output, since it's a float type.

Reply:
Oh okay, well in that case this would be fine